cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -51,10 +52,18 @@ type Actuator struct {
 	nodeDeletionBatcher *NodeDeletionBatcher
 	evictor             Evictor
 	deleteOptions       simulator.NodeDeleteOptions
+	configGetter        actuatorNodeGroupConfigGetter
+}
+
+// actuatorNodeGroupConfigGetter is an interface to limit the functions that can be used
+// from NodeGroupConfigProcessor interface
+type actuatorNodeGroupConfigGetter interface {
+	// GetIgnoreDaemonSetsUtilization returns IgnoreDaemonSetsUtilization value that should be used for a given NodeGroup.
+	GetIgnoreDaemonSetsUtilization(context *context.AutoscalingContext, nodeGroup cloudprovider.NodeGroup) (bool, error)
 }
 
 // NewActuator returns a new instance of Actuator.
-func NewActuator(ctx *context.AutoscalingContext, csr *clusterstate.ClusterStateRegistry, ndt *deletiontracker.NodeDeletionTracker, deleteOptions simulator.NodeDeleteOptions) *Actuator {
+func NewActuator(ctx *context.AutoscalingContext, csr *clusterstate.ClusterStateRegistry, ndt *deletiontracker.NodeDeletionTracker, deleteOptions simulator.NodeDeleteOptions, processors *processors.AutoscalingProcessors) *Actuator {
```

Comment:
nit: Do we need to have all processors passed to actuator? It should suffice to just have a param of type `actuatorNodeGroupConfigGetter` here.

Reply:
Thank you for the comment. Makes sense. I have updated the code.